### PR TITLE
docs/data-modeling: Adding compareDateRange example

### PR DESCRIPTION
**Pull request by EduardoGHdez**
**Check List**
- [ ] ~Tests has been run in packages where changes made if available~
- [ ] ~Linter has been run for changed code~
- [ ] ~Tests for the changes have been added if not covered yet~
- [x] Docs have been added / updated if required

**Issue Reference this PR resolves**
N/A

**Description of Changes Made (if issue reference is not provided)**
While reading the docs, it took me a few searches to figure out how to make use of the `compareDateRange` functionality through plain REST API. Just thought on adding an example for quick reference, but kept the references to the [Comparing Data over Different Time Periods](https://cube.dev/blog/comparing-data-over-different-time-periods) blogpost

### CODE REVIEW

File: docs/pages/product/data-modeling/queries.mdx [hunk 1]
```diff
@@ -92,7 +92,7 @@ The same query using the REST API syntax looks as follows:
 
 ### Ungrouped query
 
-Similarly to [regular queries](#regular-queries), ungrouped queries include
+Similarly to [regular queries](#regular-query), ungrouped queries include
```
Comment (EduardoGHdez) [inline docs/pages/product/data-modeling/queries.mdx:97]: These links weren't working on the docs 🤓

**Comment by EduardoGHdez** [comment]
Hey @igorlukanin I wonder if this is something you may be able to take a look at 😄 

Thank you in advance 🙇🏽

**Comment by EduardoGHdez** [comment]
Update: Just noticed that there were some major changes to the docs 😅. I updated the corresponding sections according to the new location of these sections 👍🏽 cc @igorlukanin

**Comment by EduardoGHdez** [comment]
> Hi @EduardoGHdez 👋
> 
> Thanks a ton for your contribution—and especially for adjusting for some later changes to docs! I really appreciate the effort!
> 
> I see what you've done with `query` and `queryType` here. However, I do think it might be quite confusing. What about "unwrapping" the REST API queries back and then maybe leaving a note that compare date range queries would also require `queryType` to be run?

Hey, @igorlukanin , thanks you input here 😄 . I just went ahead and applied your suggestions to the PR 🙌🏽 
Let me know wdyt

**Comment by igorlukanin** [comment]
Huge kudos @EduardoGHdez 🙌 I appreciate your effort!

**Pull request merged**